### ISSUE TITLE
ES6 module syntax brings in un-imported bindings

### ISSUE THREAD

**Issue by duncanmak** [Microsoft/TypeScript#3806]
Using the new ES6 module syntax with 1.5.0-beta, this compiles, but shouldn't:

```typescript
/// <reference path="lodash.d.ts" />
import { pad } from 'lodash'

console.log(pad('abc', 8));
console.log(_.pad('abc', 8)); // BAD: This shouldn't compile, but tsc doesn't catch it
```

If you use the older syntax, it works as expected:

```typescript
/// <reference path="lodash.d.ts" />
import _ = require('lodash');

console.log(pad('abc', 8)); // GOOD: This line doesn't compile, and that's correct
console.log(_.pad('abc', 8));
```

I have a runnable test case here - https://github.com/duncanmak/ts-bug

**Comment by mhegazy** [comment]
lodash.d.ts adds a global variable `_` to the namespace. in the second case, you are shadowing the global variable `_` with your local import. 

You can modify lodash.d.ts to remove the global, and the only way to import it is using the `import` syntax.

**Comment by duncanmak** [comment]
Ah, thanks @mhegazy, I suspected that it might be something like this.

So, the issue really is, how should .`d.ts` files be written? Is it as simple as not exporting global variables?

I think this issue will come up again and again when people start to use the new import syntax instead of the old one.

**Comment by mhegazy** [comment]
one thing to note, that .d.ts represent the behavior of the library. most of libraries in use today add a named global, e.g. `_` or `$`.

if you do not want to use the globals, the definitions need to be forked for modules and globals.

**Comment by vvakame** [comment]
I think separate lodash.d.ts is better way.
lodash-common.d.ts, lodash-browser.d.ts and lodash-node.d.ts. like this http://goo.gl/8cOaUl
If you want to continue this story, raise new issue in DefinitelyTyped repo. and ping me and @basarat .
